src/main/java/io/supertokens/authRecipe/AuthRecipe.java

```diff
@@ -117,76 +123,63 @@ public static void deleteUser(TenantIdentifier tenantIdentifier, Main main, Stri
-        } catch (TenantOrAppNotFoundException | BadPermissionException e) {
+            Storage storage = StorageLayer.getStorage(main);
+            deleteUser(new AppIdentifier(null, null, storage), userId);
+        } catch (BadPermissionException e) {
             throw new IllegalStateException(e);
         }
     }
 
-    private static void deleteNonAuthRecipeUser(TenantIdentifier tenantIdentifier, Main main, String userId)
-            throws StorageQueryException, TenantOrAppNotFoundException {
-        StorageLayer.getUserMetadataStorage(tenantIdentifier, main)
-                .deleteUserMetadata(tenantIdentifier.toAppIdentifier(), userId);
-        StorageLayer.getSessionStorage(tenantIdentifier, main)
-                .deleteSessionsOfUser(tenantIdentifier.toAppIdentifier(), userId);
-        StorageLayer.getEmailVerificationStorage(tenantIdentifier, main)
-                .deleteEmailVerificationUserInfo(tenantIdentifier.toAppIdentifier(), userId);
-        StorageLayer.getUserRolesStorage(tenantIdentifier, main)
-                .deleteAllRolesForUser(tenantIdentifier.toAppIdentifier(), userId);
+    private static void deleteNonAuthRecipeUser(AppIdentifier appIdentifier, String userId)
+            throws StorageQueryException {
+        appIdentifier.getUserMetadataStorage()
+                .deleteUserMetadata(appIdentifier, userId);
+        appIdentifier.getSessionStorage()
+                .deleteSessionsOfUser(appIdentifier, userId);
+        appIdentifier.getEmailVerificationStorage()
+                .deleteEmailVerificationUserInfo(appIdentifier, userId);
+        appIdentifier.getUserRolesStorage()
+                .deleteAllRolesForUser(appIdentifier, userId);
     }
 
-    private static void deleteAuthRecipeUser(TenantIdentifier tenantIdentifier, Main main, String userId)
-            throws StorageQueryException, TenantOrAppNotFoundException {
+    private static void deleteAuthRecipeUser(AppIdentifier appIdentifier, String userId)
+            throws StorageQueryException {
         // auth recipe deletions here only
-        StorageLayer.getEmailPasswordStorage(tenantIdentifier, main)
-                .deleteEmailPasswordUser(tenantIdentifier.toAppIdentifier(), userId);
-        StorageLayer.getThirdPartyStorage(tenantIdentifier, main)
-                .deleteThirdPartyUser(tenantIdentifier.toAppIdentifier(), userId);
-        StorageLayer.getPasswordlessStorage(tenantIdentifier, main)
-                .deletePasswordlessUser(tenantIdentifier.toAppIdentifier(), userId);
+        // TODO delete from app_id_to_user_id table
+        Storage storage = appIdentifier.getStorage();
+        ((EmailPasswordSQLStorage) storage).deleteEmailPasswordUser(appIdentifier, userId);
+        ((ThirdPartySQLStorage) storage).deleteThirdPartyUser(appIdentifier, userId);
+        ((PasswordlessSQLStorage) storage).deletePasswordlessUser(appIdentifier, userId);
```

Comment:
This can be done via a single function call instead of per recipe.

Reply:
Retained it, because, passwordless storage also deletes devices for the user's email/phoneNumber